src/user/user.controller.ts

```diff
@@ -1,93 +1,74 @@
+        return await this.userService.transformToReturn(user);
+    }
 
-  @Get("/verify/:code")
-  @Render("MailVerify")
-  async verifyMail(@Param('code') code: string): Promise<any> {
-    let result = await this.userService.veryfiyUser(code)
-    return
+    @Get('/profile')
+    @UseGuards(JwtAuthGuard)
+    async getProfile(@Request() req): Promise<returnUser> {
+        const user = await this.userService.findOneById(req.user.userId);
+        return await this.userService.transformToReturn(user);
+    }
 
-  }
+    @Get('/get-verify')
+    @UseGuards(JwtAuthGuard)
+    async regenerateVerify(@Request() req): Promise<void> {
+        const userData = await this.userService.parseJWTtOUsable(req.user);
+        await this.userService.createVerification(userData);
+    }
 
-  @Get('/profile')
-  @UseGuards(JwtAuthGuard)
-  getProfile(@Request() req): Promise<any> {
-    return req.user
-  }
+    @Patch('/:id')
+    @UseGuards(JwtAuthGuard)
+    async update(
+        @Param('id') id: ObjectId,
+        @Body() updateUserDto: UpdateUserDto
+    ): Promise<returnUser> {
+        const user = await this.userService.updateUser(id, updateUserDto);
+        return await this.userService.transformToReturn(user);
+    }
 
-  /**
-   * FOR TESTING
-   * @param id Object Id
-   * @param role body
-   */
-  @Patch('/testing/:id')
-  async updateRole(@Param('id') id: ObjectId, @Body() role: Role) {
-    return await this.userService.patchRole(id, role)
-  }
-
-  @Get('/getVerify')
-  @UseGuards(JwtAuthGuard)
-  async regenerateVerify(@Request() req): Promise<any> {
-    return this.userService.createVerification(await this.userService.parseJWTtOUsable(req.user))
-  }
-
-  @Patch('/:id')
-  async update(@Param('id') id: ObjectId, @Body() updateUserDto: UpdateUserDto): Promise<User> {
-    return await this.userService.updateUser(id, updateUserDto);
-  }
-
-  @Delete('/:id')
-  async remove(@Param('id') id: ObjectId): Promise<any> {
-    return await this.userService.remove(id);
-  }
-
-  @Get('/password-reset')
-  async resetPassword(@Body() userData: { userMail: string }) {
-    return await this.userService.requestResetPassword(userData.userMail)
-  }
-
-  @Get('/reset')
-  @UseGuards(JwtAuthGuard, RolesGuard)
-  @Roles(Role.Admin)
-  async getResetByUsername(@Body() userData: { userId: ObjectId }): Promise<any> {
-    return await this.userService.findReset(userData.userId)
-  }
-
-  @Get('/reset-form/:code')
-  @Render('reset')
-  async getResetForm(@Param('code') Usercode: string) {
-    return { code: Usercode, submitURL: `${process.env.HOST}/api/user/submitReset` }
-  }
-
-  @Post('/submitReset')
-  async validateReset(@Body() values: { password: string, code: string }) {
-    return await this.userService.validatePasswordReset(values.code, values.password)
-  }
+    @Delete('/:id')
+    @HttpCode(204)
+    @UseGuards(JwtAuthGuard)
+    async remove(@Param('id') id: ObjectId, @Request() req): Promise<void> {
```

Comment:
Same here we should discuss who can delete a user. Can you delete yourself?

Reply:
Yes you should be able to delete yourself => users might want to delete their account at some point